synapseadmin/roomapi.go

```diff
@@ -0,0 +1,167 @@
+// Copyright (c) 2023 Tulir Asokan
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package synapseadmin
+
+import (
+	"context"
+	"net/http"
+	"maunium.net/go/mautrix"
+	"maunium.net/go/mautrix/id"
+	"strconv"
+)
+
+type ReqListRoom struct {
+	Alias	string
+	From	int
+	Limit   int
+}
+
+func (cli *Client) NewReqListRoom() ReqListRoom {
+   req := ReqListRoom{}
+   req.Alias = ""
+   req.From = 0
+   req.Limit = 100
+   return req
+}
+
+func (req *ReqListRoom) BuildQuery() map[string]string {
+	query := map[string]string{
+		"limit": strconv.Itoa(req.Limit),
```

Comment:
Seems like it'd be easier to omit `limit` when it's 0 and get rid of the whole `NewReqListRoom` function.